lib/Zonemaster/Backend/Config.pm

```diff
@@ -29,29 +30,152 @@ Readonly my @SIG_NAME => split ' ', $Config{sig_name};
+        }
+    );
+
+Throws an exception if the given configuration file contains errors.
+
+=cut
+
+sub parse {
+    my ( $class, $text ) = @_;
+
+    my $obj = bless( {}, $class );
+
+    my $ini = Config::IniFiles->new( -file => \$text )
+      or die "Failed to parse config: " . join( '; ', @Config::IniFiles::errors ) . "\n";
+
+    # Validate, normalize, and apply default values
+    {
+        my $engine = $ini->val( 'DB', 'engine' );
+        eval {
+            $engine = $obj->check_db($engine);
+        };
+        if ($@) {
+            die "Unknown config value DB.engine: $engine\n";
+        }
+        $obj->{_DB_engine} = $engine;
+    }
+
+    $obj->{_DB_polling_interval}                                 = $ini->val( 'DB',         'polling_interval',                         undef );
+    $obj->{_MYSQL_host}                                          = $ini->val( 'MYSQL',      'host',                                     undef );
+    $obj->{_MYSQL_user}                                          = $ini->val( 'MYSQL',      'user',                                     undef );
+    $obj->{_MYSQL_password}                                      = $ini->val( 'MYSQL',      'password',                                 undef );
+    $obj->{_MYSQL_database}                                      = $ini->val( 'MYSQL',      'database',                                 undef );
+    $obj->{_POSTGRESQL_host}                                     = $ini->val( 'POSTGRESQL', 'host',                                     undef );
+    $obj->{_POSTGRESQL_user}                                     = $ini->val( 'POSTGRESQL', 'user',                                     undef );
+    $obj->{_POSTGRESQL_password}                                 = $ini->val( 'POSTGRESQL', 'password',                                 undef );
+    $obj->{_POSTGRESQL_database}                                 = $ini->val( 'POSTGRESQL', 'database',                                 undef );
+    $obj->{_SQLITE_database_file}                                = $ini->val( 'SQLITE',     'database_file',                            undef );
+    $obj->{_ZONEMASTER_max_zonemaster_execution_time}            = $ini->val( 'ZONEMASTER', 'max_zonemaster_execution_time',            '600' );
+    $obj->{_ZONEMASTER_maximal_number_of_retries}                = $ini->val( 'ZONEMASTER', 'maximal_number_of_retries',                '0' );
+    $obj->{_ZONEMASTER_number_of_processes_for_frontend_testing} = $ini->val( 'ZONEMASTER', 'number_of_processes_for_frontend_testing', undef );
+    $obj->{_ZONEMASTER_number_of_processes_for_batch_testing}    = $ini->val( 'ZONEMASTER', 'number_of_processes_for_batch_testing',    undef );
+    $obj->{_ZONEMASTER_lock_on_queue}                            = $ini->val( 'ZONEMASTER', 'lock_on_queue',                            undef );
+    $obj->{_ZONEMASTER_age_reuse_previous_test}                  = $ini->val( 'ZONEMASTER', 'age_reuse_previous_test',                  '600' );
+
```

Comment:
Some of these should be a non-negative integer, unless undefined. Could this be checked here?

Reply:
Yes, I'll add validations for that in a later PR.

Reply:
Ok.